### PR TITLE
Augment documentation of VectorizedArray

### PR DESCRIPTION
**Pull request by kronbichler**
I also realized that we can remove some `NumberType` specializations now that `VectorizedArray` supports `operator=(scalar)` from #8426.

### CODE REVIEW

File: include/deal.II/base/vectorization.h [hunk 1]
```diff
@@ -127,19 +92,21 @@ struct EnableIfScalar<VectorizedArray<Number, width>>
...
- * sixteen floats are used.
+ * operations. For floats and doubles, an array of numbers are packed together
+ * with the goal to be processed in a single-instruction/multiple-data (SIMD)
+ * fashion. In the SIMD context, the elements of such a short vector are often
+ * called lanes. The number of elements packed together, i.e., the number of
+ * lanes, depends on the computer system and compiler flags that are used for
+ * compilation of deal.II. The fundamental idea of these packed data types is
+ * to use one single CPU instruction to perform arithmetic operations on the
+ * whole array using the processor's vector (SIMD) units. Most computer
+ * systems by 2010 standards will use an array of two doubles and four floats,
```
Comment (tjhei) [inline include/deal.II/base/vectorization.h]:
```suggestion
 * systems by 2010 standards will use an array of two doubles or four floats,
```

File: include/deal.II/base/vectorization.h [hunk 2]
```diff
@@ -127,19 +92,21 @@ struct EnableIfScalar<VectorizedArray<Number, width>>
...
+ * respectively (this corresponds to the SSE/SSE2 data sets) when compiling
+ * deal.II on 64-bit operating systems. On Intel Sandy Bridge processors and
+ * newer or AMD Bulldozer processors and newer, four doubles and eight floats
```
Comment (tjhei) [inline include/deal.II/base/vectorization.h]:
```suggestion
 * newer or AMD Bulldozer processors and newer, four doubles or eight floats
```

File: include/deal.II/base/vectorization.h [hunk 3]
```diff
@@ -127,19 +92,21 @@ struct EnableIfScalar<VectorizedArray<Number, width>>
...
+ * are used when deal.II is configured e.g. using gcc with --with-cpu=native
```
Comment (tjhei) [inline include/deal.II/base/vectorization.h]:
```suggestion
 * are used when deal.II is configured using gcc with --with-cpu=native
```

File: include/deal.II/base/vectorization.h [hunk 4]
```diff
@@ -127,19 +92,21 @@ struct EnableIfScalar<VectorizedArray<Number, width>>
...
+ * or --with- cpu=corei7-avx. On compilations with AVX-512 support (e.g.,
```
Comment (tjhei) [inline include/deal.II/base/vectorization.h]:
```suggestion
 * or --with-cpu=corei7-avx. On compilations with AVX-512 support (e.g.,
```

File: include/deal.II/base/vectorization.h [hunk 5]
```diff
@@ -127,19 +92,21 @@ struct EnableIfScalar<VectorizedArray<Number, width>>
...
+ * Intel Skylake Server from 2017), eight doubles and sixteen floats are used.
```
Comment (tjhei) [inline include/deal.II/base/vectorization.h]:
```suggestion
 * Intel Skylake Server from 2017), eight doubles or sixteen floats are used.
```

File: include/deal.II/base/vectorization.h [hunk 6]
```diff
@@ -368,18 +337,20 @@ class VectorizedArray
...
-   * written by this function will need to query main memory, which is slower
-   * than an access from a cache both latency-wise and throughput-wise. Thus,
-   * this command should only be used for large stores that will collectively
-   * not fit into caches, as performance will be degraded otherwise. For a
-   * typical use case, see also <a
-   * href="https://blogs.fau.de/hager/archives/2103">this blog article</a>.
+   * overwritten (meaning that the memory subsystem makes sure that
+   * consecutive stores that together span a cache line are merged, and
+   * appropriately handling the case where only part of a cache line is
+   * written), so no need to first read the "remainder" of the cache
```
Comment (tjhei) [inline include/deal.II/base/vectorization.h]:
```suggestion
   * written), so there is no need to first read the "remainder" of the cache
```

**Comment by kronbichler** [comment]
Fixed.

**Pull request merged**